ci/jenkinsfiles/build.groovy

```diff
@@ -17,46 +17,103 @@
 *     Abdoul BA <aba@nuxeo.com>
 */
 
+/* Using a version specifier, such as branch, tag, etc */
+@Library('nuxeo-napps-tools@improv-NXBT-3447-napps-shared-libs') _
+
 def appName = 'nuxeo-coldstorage'
-def pipelineLib
 def repositoryUrl = 'https://github.com/nuxeo/nuxeo-coldstorage/'
 
-properties([
-  [
-    $class: 'BuildDiscarderProperty',
-    strategy: [
-      $class: 'LogRotator',
-      daysToKeepStr: '15', numToKeepStr: '10',
-      artifactNumToKeepStr: '5'
-    ]
-  ],
-  [
-    $class: 'GithubProjectProperty', projectUrlStr: repositoryUrl
-  ],
-  disableConcurrentBuilds(),
-])
+String getVersion() {
+  return nxNapps.isPullRequest() ? nxNapps.getPullRequestVersion() : getReleaseVersion()
+}
+
+String getReleaseVersion() {
+  String preid = 'rc'
+  String nextPromotion = readMavenPom().getVersion().replace('-SNAPSHOT', '')
+  String version = "${nextPromotion}-${preid}.0" // first version ever
 
-void setGitHubBuildStatus(String context, String message, String state, String gitRepo) {
-  if ( env.DRY_RUN != 'true' && ENABLE_GITHUB_STATUS == 'true') {
-    step([
-      $class: 'GitHubCommitStatusSetter',
-      reposSource: [$class: 'ManuallyEnteredRepositorySource', url: gitRepo],
-      contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
-      statusResultSource: [
-        $class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]
-      ],
-    ])
+  // find the latest tag if any
+  sh "git fetch origin 'refs/tags/v${nextPromotion}*:refs/tags/v${nextPromotion}*'"
+  String tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${nextPromotion}*' | tail -1 | tr -d '\n'")
+  if (tag) {
+    container('maven') {
+      version = sh(returnStdout: true, script: "npx --ignore-existing semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+def runBackEndUnitTests() {
+  return {
+    stage('BackEnd') {
+      container('maven-default') {
+        script {
+          gitHubBuildStatus.set('utests/backend')
+          try {
+            echo '''
+              ----------------------------------------
+              Run BackEnd Unit tests
+              ----------------------------------------
+            '''
+            sh """
+              cd ${BACKEND_FOLDER}
+              mvn ${MAVEN_ARGS} -V -T0.8C test
+            """
+          } catch (err) {
+            throw err
+          } finally {
+            junit allowEmptyResults: true, testResults: "**/target/surefire-reports/*.xml"
+            gitHubBuildStatus.set('utests/backend')
+          }
```

Comment:
same than [Here](https://github.com/nuxeo/nuxeo-coldstorage/pull/29/commits/1bfcbbfb94686f4ff5108f0700f73ed4c1cbeac3#r609835870) about the renaming or only use `tests/backend` I don't know what Platform is using.

Reply:
Platform Team use the old syntax

`setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'PENDING')`

env can be mongodb or postgresql